FirebaseSwiftUI/FirebaseGoogleSwiftUI/Sources/Services/GoogleProviderSwift.swift

```diff
@@ -0,0 +1,61 @@
+@preconcurrency import FirebaseAuth
+import FirebaseAuthSwiftUI
+import GoogleSignIn
+
+let kGoogleUserInfoEmailScope = "https://www.googleapis.com/auth/userinfo.email"
+let kGoogleUserInfoProfileScope = "https://www.googleapis.com/auth/userinfo.profile"
+let kDefaultScopes = [kGoogleUserInfoEmailScope, kGoogleUserInfoProfileScope]
+
+public enum GoogleProviderError: Error {
+  case rootViewControllerNotFound(String)
+  case authenticationToken(String)
+  case user(String)
+}
+
+public class GoogleProviderSwift: @preconcurrency GoogleProviderProtocol {
```

Comment:
I'm not super happy with the `Swift` suffix. Also, can we include `Auth` in the name? Same for the protocol(s).

This applies to all providers.